### PR TITLE
Introduce tracking-only reco, validation, and dqm sequences

### PR DESCRIPTION
**Pull request by makortel**
This PR is request for comments for introduction of tracking-only reconstruction, validation, and dqm sequences allowing their use with cmsDriver as

```
cmsDriver.py step3 ... -s RAW2DIGI,RECO:reconstruction_trackingOnly,VALIDATION:@trackingOnlyValidation,DQM:@trackingOnlyDQM
cmsDriver.py step4 ... -s HARVESTING:@trackingOnlyValidation+@trackingOnlyDQM
```

If accepted, would it be possible to add one workflow with it in the IBs? (even though we probably will exercise this ourselves frequently, a specific test would help to ensure that the workflow stays heathy)

Tested in CMSSW_8_0_X_2015-11-22-1100, no changes expected.

@rovere @VinInn

### CODE REVIEW

File: Validation/RecoTrack/python/TrackValidation_cff.py [hunk 1]
```diff
@@ -282,24 +282,51 @@ def _addSelectorsByOriginalAlgoMask(modules, midfix, algoParam):
...
+trackValidatorsTrackingOnly.replace(trackValidatorStandalone, trackValidatorTrackingOnly)
+
+tracksValidationTrackingOnly = cms.Sequence(
+    trackValidatorsTrackingOnly
+)
 
 
 ### 'slim' sequences that only depend on track and tracking particle collections
-tracksValidationSelectorsSlim = tracksValidationSelectors.copyAndExclude([cutsRecoTracksBtvLike,ak4JetTracksAssociatorExplicitAll,cutsRecoTracksAK4PFJets])
+tracksValidationSelectorsSlim = tracksValidationSelectorsTrackingOnly.copyAndExclude([cutsRecoTracksBtvLike])
```
Comment (makortel) [inline Validation/RecoTrack/python/TrackValidation_cff.py]: @lveldere After this, the only difference between "trackingOnly" and "slim" would be that the former includes "BTV-like" track selection while the latter does not. I don't mind keeping "slim", but thought to ask anyway if keeping it is motivated?

**Comment by VinInn** [comment]
@cmsbuild , please test

**Comment by slava77** [comment]
from reco perspective, a separate sequence in Configuration/StandardSequences/python/Reconstruction_cff.py looks OK, rather technical.
@boudoul @hengne please follow up about a relval workflow here.

File: SLHCUpgradeSimulations/Configuration/python/phase1TkCustoms.py [hunk 2]
```diff
@@ -310,8 +310,8 @@ def customise_Reco(process,pileup):
     # add the correct tracking back in
     process.load("RecoTracker.Configuration.RecoTrackerPhase1PU"+str(nPU)+"_cff")
 
-    process.globalreco.insert(itIndex,process.trackingGlobalReco)
-    process.reconstruction.insert(grIndex,process.globalreco)
+    process.globalreco_tracking.insert(itIndex,process.trackingGlobalReco)
```
Comment (makortel) [inline SLHCUpgradeSimulations/Configuration/python/phase1TkCustoms.py]: @mark-grimes I guess these changes should not interfere with the phase1 era migration, but FYI nevertheless.

**Comment by civanch** [comment]
+1

File: Validation/RecoTrack/python/TrackValidation_cff.py (hunk 1 above)
Comment (lveldere) [inline Validation/RecoTrack/python/TrackValidation_cff.py]: Hi Matti

I agree with you.

Thanks

Lukas


On Mon, Nov 23, 2015 at 2:58 PM, Matti Kortelainen <notifications@github.com

**Comment by cvuosalo** [comment]
+1

For #12542 91474dcea88f07c6404e264137731c27059b17ba

Adding tracking-only reconstruction, validation, and DQM sequences. 

Reco has no objections to the new sequences. Jenkins tests show no significant differences.

**Comment by makortel** [comment]
Could DQM please comment? @deguio @vanbesien

**Comment by deguio** [comment]
ciao @makortel 
I have only few comments:
- probably for uniformity you should update the entry in the DQM matrix to @trackingOnlyDQM
- all the validation modules that don't need to access the trigger results should be moved to the preValidationSequence.

for the rest the changes are good for DQM and Valid.
cheers,
F.

**Comment by makortel** [comment]
Thanks @deguio. I changed the name to `@trackingOnlyDQM`, but did not shuffle the sequences yet. I started to think if it would be more clear to do that in a subsequent PR, because I would like to do it on both offline and HLT tracking and vertex validation sequences (and maybe including all those here would be out of scope of this PR?). I had probably misunderstood the separation between prevalidation and validation sequences (I had thought producers and analyzers), if the requirement for a module to be in validation sequence is only the trigger results, I think I can move all our modules to the prevalidation sequence (we need only the objects, not decisions on paths).

Would it be sufficient to address the sequences in a separate PR?

**Comment by deguio** [comment]
please test

**Comment by deguio** [comment]
+1
fine, thanks.
the validation sequence is run in an EndPath to make the modules which need it, to be able to access trigger results which are produced and injected into the event after Paths and before EndPaths. on the other hand modules in the EndPaths could not be freely reshuffled when running unscheduled. they will remain confined there.

**Comment by civanch** [comment]
+1

**Comment by cvuosalo** [comment]
+1

For #12542 103f3aeb799eca4f033efb77648a4b3f5657d448

Second approval after tiny code change.

New Jenkins tests still show no significant differences.

**Comment by smuzaffar** [comment]
looks like this PR has caused a unit test to fail in 80X IBs https://cmssdt.cern.ch/SDT/cgi-bin/buildlogs/slc6_amd64_gcc493/CMSSW_8_0_X_2015-12-09-0000/unitTestLogs/Configuration/DataProcessing
You can reproduce it by doing
```
scram project  CMSSW_8_0_X_2015-12-05-2300
cd CMSSW_8_0_X_2015-12-05-2300
cmsenv
git cms-merge-topic 12542
git cms-addpkg Configuration/DataProcessing
scram b -v -j 20
cmsenv
scram b runtests_TestConfigDP
```

**Comment by makortel** [comment]
Investigating

**Comment by makortel** [comment]
@smuzaffar The fix is in #12730. Thanks for reporting.

**Comment by smuzaffar** [comment]
thanks

**Pull request merged**